src/irmin/irmin.mli

```diff
@@ -951,6 +951,9 @@ module type ATOMIC_WRITE_STORE = sig
   val close : t -> unit Lwt.t
   (** [close t] frees up all the resources associated to [t]. Any
       operations run on a closed store will raise {!Closed}. *)
+
+  val clear : t -> unit Lwt.t
```

Comment:
Is this (different from) / (better than) than just creating a new store and throwing away the old one?

Reply:
`clear` is my next PR to extract from the layered store. 
I'm not sure what is best. Throwing away the old store means closing it, and leaving it to the user to remove all disk files generated by the store. Whereas `clear` handles it for the user.

Reply:
We could have a `close_and_clear` operation on the discarded store, if your concern is only user-responsibility for destroying the underlying resources. I brought it up because `clear` is that an imperative API (where I like things to be as functional as possible :slightly_smiling_face:), but looking more closely the rest of this signature is imperative anyway.

Reply:
I think I prefer having a separate `clear` and `close`, instead of having two `close` operations. Alternatively, in Irmin-pack we have a fresh flag, for reopening a store and we could propagate it for all backend.

Reply:
Let's stick with `clear` and `close` then. I already dislike the `fresh` flag enough where it is :slightly_smiling_face: